src/routes/posts/+page.server.ts

```diff
@@ -51,3 +55,47 @@ export const load: PageServerLoad = async ({ platform, url }) => {
 		end
 	};
 };
+
+export const actions: Actions = {
+	update: async ({ request, platform }) => {
+		if (platform === undefined) {
+			error(400, { message: "Platform undefined"})
+		}
+
+		const data = await request.formData();
+
+		const images_without_alt_text: number = await fetch(
+			`https://michigandaily.com/wp-json/tmd/v1/posts/?ids=${data.get('id')}&content=true&image=true`
+		)
+			.then((resp) => {
+				if (!resp.ok) {
+					throw Error(`${resp.status} ${resp.statusText}`);
+				}
+				return resp.json() as Promise<Array<Article>>;
+			})
+			.then((data) => {
+				return parseContent(data[0].image, data[0].content).length;
+			})
+			.catch((msg) => {
+				error(400, msg);
+			});
+
+		const response = await platform.env.DB.prepare(
+			`UPDATE articles 
+			SET images_published_with_alt_text = images_published - ? 
+			WHERE aid == ? `
+		)
+			.bind(images_without_alt_text, data.get('id'))
+			.run();
+
+		if (response.error) {
+			fail(400, { message: response.error });
+		}
+
+		if (response.meta.rows_written < 1) {
+			fail(400, { message: "Failed to update article"})
+		}
```

Comment:
I believe you need to return the failure. Personally, I think it's weird that you don't have to return for `error` or `redirect` but you have to for `fail` but that's how it was designed.